editor/effects.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import { parse, stringify } from 'querystring';
+
+/**
+ * WordPress dependencies
+ */
+import { getBlockSettings, switchToBlockType } from 'blocks';
+
+/**
+ * Internal dependencies
+ */
+import { focusBlock, replaceBlocks } from './actions';
+
+export default {
+	REQUEST_POST_UPDATE( action, store ) {
```

Comment:
>Just a crazy idea: What if we write the effects as thunks to make them "compatible" with the way the thunk middleware works? `REQUEST_POST_UPDATE = ( action ) => ( dispatch ) => {}` 

Funny you mention it, a similar idea had been on my mind yesterday for a reason I don't recall. In my case thinking to align with the middleware signature `( store ) => ( action ) => ( next )` to allow these to act as mini-middlewares of sorts. In the end I didn't find it particularly compelling outside the consistency point, and in fact I've been trying to push `refx` more in a direction of encouraging synchronous flows, discouraging the `store` argument as much as possible (not always possible).

See also:

- [v2.0.0 Changelog notes](https://github.com/aduth/refx/blob/master/CHANGELOG.md)
- [Example in practice, using a separate requests middleware](https://github.com/aduth/dones/blob/fac919a146053ffe1f539055ae5e4ffa4b82774f/src/state/dones/effects.js#L14-L65)